privacyidea/api/lib/postpolicy.py

```diff
@@ -248,6 +248,47 @@ def check_serial(request, response):
     return response
 
 
+def check_tokeninfo(request, response):
+    """
+    This policy function is used as a decorator for the validate API.
+    It checks after a successful authentication if the token has a matching
+    tokeninfo field. If it does not match, authorization is denied. Then
+    a PolicyException is raised.
+
+    :param response: The response of the decorated function
+    :type response: Response object
+    :return: A new modified response
+    """
+    content = json.loads(response.data)
+    policy_object = g.policy_object
+    serial = content.get("detail", {}).get("serial")
+    if serial:
+        tokens = get_tokens(serial=serial)
+        if len(tokens) == 1:
+            token_obj = tokens[0]
+            tokeninfos_pol = policy_object.get_action_values(
+                ACTION.TOKENINFO,
+                scope=SCOPE.AUTHZ,
+                client=g.client_ip,
+                allow_white_space_in_action=True)
+            for tokeninfo_pol in tokeninfos_pol:
+                try:
+                    key, regex, _r = tokeninfo_pol.split("/")
+                    value = token_obj.get_tokeninfo(key, "")
+                    if re.search(regex, value):
```

Comment:
``re.search`` looks for a match [anywhere in the string](https://docs.python.org/2/library/re.html#re.search). Maybe we should use ``re.match`` instead?

Reply:
This is intended. You can add a caret ^ to the regex, to search from the start. Thus ``search`` is more flexible than ``match``.

Reply:
Oops yes, obviously. I was confusing myself. :)